demo/src/main/java/eu/kevin/demo/main/MainViewModel.kt

```diff
@@ -17,71 +29,160 @@ import kotlinx.coroutines.flow.receiveAsFlow
 
-    private val kevinAuthClient = KevinApiClientFactory(
-        baseUrl = BuildConfig.KEVIN_API_URL,
-        "",
-        timeout = 120000,
-        logLevel = LogLevel.NONE
-    ).createClient(null)
-
-    private val _viewState = MutableStateFlow<MainViewState>(MainViewState.Loading(false))
+    private val _viewState = MutableStateFlow(MainViewState())
     private val _viewAction = Channel<MainViewAction>(Channel.BUFFERED)
 
     val viewState: StateFlow<MainViewState> = _viewState
     val viewAction = _viewAction.receiveAsFlow()
 
-    fun initializeAccountLinking() {
-        viewModelScope.launch(Dispatchers.IO) {
-            _viewState.update { MainViewState.Loading(true) }
-            try {
-                val state = kevinAuthClient.getAuthState(
-                    InitiateAuthenticationRequest(
-                        scopes = listOf(AuthenticationScope.PAYMENTS.value)
-                    )
+    init {
+        loadCreditors(_viewState.value.selectedCountry)
+    }
+
+    fun onCreditorSelected(creditor: CreditorListItem) {
+        _viewState.update {
+            it.copy(
+                creditors = it.creditors.map {
+                    it.copy(isSelected = it == creditor)
+                },
+                selectedCreditor = creditor,
+                loadingState = LoadingState.Loading(false)
+            )
+        }
+    }
+
+    fun onCountrySelected(iso: String) {
+        _viewState.update {
+            it.copy(
+                selectedCountry = iso,
+                loadingState = LoadingState.Loading(false)
+            )
+        }
+        loadCreditors(iso)
+    }
+
+    fun openCountrySelection() {
+        val config = CountrySelectionFragmentConfiguration(
+            _viewState.value.selectedCountry
+        )
+        GlobalRouter.pushModalFragment(CountrySelectionContract.getFragment(config))
+    }
+
+    fun onAmountChanged(amount: String) {
+        _viewState.update {
+            it.copy(
+                buttonText = amount.toRepresentableBigDecimal() ?: "0.00",
+                loadingState = LoadingState.Loading(false)
+            )
+        }
+    }
+
+    fun donate(donationRequest: DonationRequest) {
+        val emailValidationResult = EmailValidator.validate(donationRequest.email)
+        val amountValidationResult = AmountValidator.validate(donationRequest.amount)
+
+        _viewAction.trySend(
+            MainViewAction.ShowFieldValidations(
+                emailValidationResult,
+                amountValidationResult,
+                donationRequest.isTermsAccepted
+            )
+        )
+
+        if (emailValidationResult.isValid() && amountValidationResult.isValid() && donationRequest.isTermsAccepted) {
+            initiatePayment(
+                InitiateDonationRequest(
+                    email = donationRequest.email,
+                    amount = donationRequest.amount,
+                    iban = viewState.value.selectedCreditor?.iban ?: "",
```

Comment:
just show error if creditor is null for some reason. There is no reason to make network call with invalid data